### PR TITLE
feature: Update home dashboard

### PR DESCRIPTION
**Pull request by Mugenor**
## Summary

Update the homepage UI to:
- Use horisontal bar chart for tickets by impact
- Use incoming-vs-resolved chart instead of tickets by status
image

### CODE REVIEW

File: ui/src/lib/dateRange.ts [hunk 1]
```diff
@@ -25,6 +27,39 @@ type DateRangeOptions<T extends string> = {
...
+    }
+
+    const date = value instanceof Date ? value : new Date(value)
+    return Number.isNaN(date.getTime()) ? undefined : date
+}
+
+export function formatTimeBucketLabel(value: string | Date, resolution: TimeBucketResolution): string {
+    const date = toValidDate(value)
+    if (!date) {
+        return ''
```
Comment (pyoio) [inline ui/src/lib/dateRange.ts]: ### ⚠️ Critical: Silent empty label on unparseable dates

Returning `''` here silently corrupts chart X-axis labels when the backend sends an unexpected timestamp format. The user sees data points with blank labels and no indication of what went wrong.

Consider logging a warning and falling back to the raw string so at least something is visible:

```ts
if (!date) {
    console.warn(`formatTimeBucketLabel: unable to parse date value: ${String(value)}`)
    return String(value)
}
```

File: ui/src/components/stats/stats.tsx [hunk 2]
```diff
@@ -1,20 +1,197 @@
...
+            [normalizeTeamKey(team.name), team.code],
+            [normalizeTeamKey(team.label), team.code],
+            [normalizeTeamKey(team.code), team.code],
+        ])
+    )
+
+    return effectiveTeams
+        .filter(team => team !== TEAM_SCOPE.NO_TEAMS)
+        .map(team => teamCodeByScopeKey.get(normalizeTeamKey(team)))
+        .filter((team): team is string => !!team)
```
Comment (pyoio) [inline ui/src/components/stats/stats.tsx:68]: ### ⚠️ Critical: Unresolvable teams are silently dropped

If a team name/label doesn't match any entry in `userTeams` (e.g., normalization mismatch, race condition during team list refresh), it is silently excluded from the backend query. The chart then shows incomplete data with no visual indicator.

This function also has no unit test coverage -- edge cases like mismatched casing, empty `userTeams`, or duplicate entries aren't exercised.

Suggestion: log when a team fails to resolve so data discrepancies are traceable:

```ts
.map(team => {
    const code = teamCodeByScopeKey.get(normalizeTeamKey(team))
    if (!code) {
        console.warn(`getIncomingResolvedTeamCodes: could not resolve team "${team}" to a code`)
    }
    return code
})
```

File: api/service/src/main/java/com/coreeng/supportbot/dashboard/JdbcDashboardRepository.java [hunk 3]
```diff
@@ -289,32 +291,75 @@ SELECT NOW() - first_open_ts::timestamptz AS age
...
+                .fetch(r -> new IncomingVsResolved(
+                        r.get("time_bucket", java.sql.Timestamp.class)
+                                .toInstant()
+                                .toString(),
+                        nullToZero(r.get("incoming", Long.class)),
+                        nullToZero(r.get("resolved", Long.class))));
+        return new IncomingVsResolvedRate(bucketing.granularity(), data);
+    }
+
+    private @Nullable LocalDate resolveIncomingVsResolvedStart(
```
Comment (pyoio) [inline api/service/src/main/java/com/coreeng/supportbot/dashboard/JdbcDashboardRepository.java]: ### 📒 Important: `resolveIncomingVsResolvedStart` has no test coverage

This private method has 3 non-obvious branches:
1. `!allTime` with/without `dateFrom` (7-day default)
2. `allTime` with explicit `dateFrom`
3. `allTime` without `dateFrom` (falls through to `findEarliestIncomingVsResolvedActivity`)

A bug here (e.g., accidentally returning `null` for `!allTime` when `dateFrom` is null) would cause the dashboard to silently return empty data instead of the 7-day lookback, and no test catches it.

Consider extracting to package-private for direct testing, or adding an integration test that mocks the `DSLContext` to verify each branch.

File: api/service/src/main/java/com/coreeng/supportbot/dashboard/JdbcDashboardRepository.java [hunk 4]
```diff
@@ -345,31 +392,153 @@ resolved_counts AS (
...
+    }
+
+    record IncomingVsResolvedBucketing(
+            IncomingVsResolvedQuery.Granularity granularity, String truncUnit, String interval) {
+
+        private static final int LEGACY_DAILY_BUCKET_THRESHOLD_DAYS = 60;
+        private static final int MAX_HOURLY_RANGE_DAYS = 4;
+        private static final int MAX_DAILY_RANGE_MONTHS = 2;
+
+        static IncomingVsResolvedBucketing forEmptyResult(IncomingVsResolvedQuery.@Nullable Granularity granularity) {
```
Comment (pyoio) [inline api/service/src/main/java/com/coreeng/supportbot/dashboard/JdbcDashboardRepository.java]: ### 📒 Important: `forEmptyResult` edge cases are untested

`forEmptyResult(AUTO)` throws `IllegalStateException` but no test covers this path. Also `forEmptyResult(null)` (which should return `daily()` per line 465-466) is untested.

The `AUTO` case on line 472 is also dead code -- it's already handled by the guard on line 465. The compiler requires the exhaustive switch, but a reader may not realize the branch is unreachable.

Suggestion: add tests for `forEmptyResult(null)` and `forEmptyResult(AUTO)` to lock in the expected behavior.

File: api/service/src/main/java/com/coreeng/supportbot/dashboard/DashboardRepository.java [hunk 5]
```diff
@@ -60,6 +60,13 @@ record UnresolvedTicketAges(String p50, String p90) {}
 
     record IncomingVsResolved(String time, long incoming, long resolved) {}
 
+    record IncomingVsResolvedRate(IncomingVsResolvedQuery.Granularity granularity, List<IncomingVsResolved> data) {
```
Comment (pyoio) [inline api/service/src/main/java/com/coreeng/supportbot/dashboard/DashboardRepository.java]: ### 📒 Important: `AUTO` granularity can leak into response type

This record uses `IncomingVsResolvedQuery.Granularity` which includes `AUTO`, but `AUTO` should never appear in a response -- it's a request-time sentinel that must be resolved before SQL execution.

Interestingly, the frontend already handles this correctly by splitting `IncomingVsResolvedGranularity` (response: `HOUR | DAY | WEEK`) from `IncomingVsResolvedRequestGranularity` (request: `AUTO | HOUR | DAY | WEEK`). Consider mirroring this on the backend with a response-specific enum to make it structurally impossible to return `AUTO`.

File: ui/src/lib/incomingVsResolved.ts [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+export function timeBucketResolutionForIncomingVsResolvedGranularity(
+    granularity?: IncomingVsResolvedGranularity
+): TimeBucketResolution {
+    switch (granularity) {
+        case 'HOUR':
+            return 'hour'
+        case 'WEEK':
+            return 'week'
+        case 'DAY':
+        default:
```
Comment (pyoio) [inline ui/src/lib/incomingVsResolved.ts:13]: ### 📒 Important: Silent fallback to `'day'` on unknown granularity

The `default` case silently maps `undefined` and any future backend enum value (e.g., `'MONTH'`) to `'day'`. If the backend adds a new granularity, the frontend would format all labels incorrectly with no indication.

Suggestion:
```ts
default:
    if (granularity !== undefined && granularity !== 'DAY') {
        console.warn(`Unknown IncomingVsResolvedGranularity: ${granularity}, defaulting to day`)
    }
    return 'day'
```

File: api/service/src/main/java/com/coreeng/supportbot/dashboard/JdbcDashboardRepository.java [hunk 7]
```diff
@@ -289,32 +291,75 @@ SELECT NOW() - first_open_ts::timestamptz AS age
...
-                : "date_trunc('hour', last_closed_ts::timestamptz)";
-        String interval = useDailyInterval ? "'1 day'::interval" : "'1 hour'::interval";
-
-        String startStr = start.format(ISO_DATE);
-        String endStr = end.format(ISO_DATE);
+    public IncomingVsResolvedRate getIncomingVsResolvedRate(IncomingVsResolvedQuery query) {
+        LocalDate end = query.dateTo() != null ? query.dateTo() : LocalDate.now(ZoneId.systemDefault());
+        List<String> filteredTeams = normalizeTeams(query.teams());
+        LocalDate start = resolveIncomingVsResolvedStart(query.dateFrom(), end, query.allTime(), filteredTeams);
+        IncomingVsResolvedBucketing bucketing = start == null || start.isAfter(end)
```
Comment (pyoio) [inline api/service/src/main/java/com/coreeng/supportbot/dashboard/JdbcDashboardRepository.java]: ### 📒 Important: Redundant null check creates a throwaway object

`start == null || start.isAfter(end)` is evaluated here to select the bucketing strategy, and again on line 302 for the early return. When the condition is true, the `IncomingVsResolvedBucketing` created by `forEmptyResult` on line 299 is only used for its `.granularity()` and then discarded.

Consider restructuring as a single early-return guard:
```java
if (start == null || start.isAfter(end)) {
    Granularity g = (query.granularity() == null || query.granularity() == AUTO)
            ? DAY : query.granularity();
    return new IncomingVsResolvedRate(g, List.of());
}
IncomingVsResolvedBucketing bucketing = IncomingVsResolvedBucketing.forDateRange(start, end, query.granularity());
```

File: ui/src/components/stats/stats.tsx [hunk 8]
```diff
@@ -1,20 +1,197 @@
...
+}: {
+    hasNoTeamScope: boolean
+    isLoading: boolean
+    error: Error | null
+    data: IncomingVsResolvedRatePoint[]
+}): JSX.Element {
+    const emptyMessage = hasNoTeamScope
+        ? 'Incoming and resolved ticket activity cannot be shown without team access.'
+        : 'No incoming or resolved ticket activity for the selected date range.'
+    if (hasNoTeamScope) {
```
Comment (pyoio) [inline ui/src/components/stats/stats.tsx]: ### 📒 Important: Duplicated `TimeSeriesChart` rendering

The `hasNoTeamScope` branch (lines 80-92) and the normal branch (lines 113-125) render nearly identical `<TimeSeriesChart>` elements. The only difference is `data={[]}` vs `data={data}`. Since `TimeSeriesChart` already handles empty data by showing `emptyMessage`, both branches can be collapsed:

```tsx
if (error) { return <error div> }
if (isLoading) { return <loading div> }
return (
    <TimeSeriesChart
        ...
        data={hasNoTeamScope ? [] : data}
        emptyMessage={emptyMessage}
    />
)
```

This eliminates ~15 lines of duplication. Also consider making `renderIncomingResolvedChart` a proper React component (`<IncomingResolvedChart />`) for consistency with `DateFilterControls` and to support future hooks/DevTools visibility.

**Review by pyoio — COMMENTED** [review]
## PR Review Summary

Nice work moving the bucketing logic to the backend and replacing the pie charts with more informative time series and bar charts. The parameterized SQL is a solid security improvement over the old `String.format` approach, and the frontend/backend type split for granularity (`IncomingVsResolvedGranularity` vs `IncomingVsResolvedRequestGranularity`) is textbook.

Below are findings from a comprehensive review covering code quality, error handling, test coverage, type design, comments, and simplification.

### Issue not attached to a diff line

**`ResolutionSLASection.tsx:41-42` -- Contradictory comments**

Line 41 says `// Create time-based buckets for better visualization` while line 42 says `// Already bucketed by backend; just pass through (fallback to empty)`. These contradict each other -- the code only passes through. Line 41 is misleading. Consider removing it or rewriting both as:
```ts
// Backend provides pre-bucketed data; just map to the TimeBucket shape (fallback to empty).
```

---
_Review generated with [Claude Code](https://claude.ai/code)_

File: ui/src/lib/dateRange.ts (hunk 1 above)
Comment (Mugenor) [reply ui/src/lib/dateRange.ts]: Fixed: warn and fall back to the raw value instead of an empty label.

File: ui/src/components/stats/stats.tsx (hunk 2 above)
Comment (Mugenor) [reply ui/src/components/stats/stats.tsx:68]: Fixed: unresolved teams now warn, and the edge cases are covered by tests.

File: api/service/src/main/java/com/coreeng/supportbot/dashboard/JdbcDashboardRepository.java (hunk 3 above)
Comment (Mugenor) [reply api/service/src/main/java/com/coreeng/supportbot/dashboard/JdbcDashboardRepository.java]: Covered: extracted the start-resolution logic and added direct branch tests.

File: api/service/src/main/java/com/coreeng/supportbot/dashboard/JdbcDashboardRepository.java (hunk 4 above)
Comment (Mugenor) [reply api/service/src/main/java/com/coreeng/supportbot/dashboard/JdbcDashboardRepository.java]: Covered: replaced the empty-result path with direct granularity resolution and added focused tests.

File: api/service/src/main/java/com/coreeng/supportbot/dashboard/DashboardRepository.java (hunk 5 above)
Comment (Mugenor) [reply api/service/src/main/java/com/coreeng/supportbot/dashboard/DashboardRepository.java]: Fixed: request and response granularities now use separate enums, so AUTO cannot leak into the response.

File: ui/src/lib/incomingVsResolved.ts (hunk 6 above)
Comment (Mugenor) [reply ui/src/lib/incomingVsResolved.ts:13]: Fixed: unknown backend granularities now warn before defaulting to day.

File: api/service/src/main/java/com/coreeng/supportbot/dashboard/JdbcDashboardRepository.java (hunk 7 above)
Comment (Mugenor) [reply api/service/src/main/java/com/coreeng/supportbot/dashboard/JdbcDashboardRepository.java]: Fixed: the early return now computes empty-result granularity directly, with no throwaway bucketing object.

File: ui/src/components/stats/stats.tsx (hunk 8 above)
Comment (Mugenor) [reply ui/src/components/stats/stats.tsx]: Fixed: collapsed the duplicate TimeSeriesChart rendering into a single path.

**Review by pyoio — APPROVED** [review]
All review feedback has been addressed in the follow-up commit. LGTM — nice work.
